### PR TITLE
[PDI-19777] Bug fixes by integrating Rest.java changes submitted by r…

### PR DESCRIPTION
**Pull request by niehusa**
…eporter

### CODE REVIEW

File: engine/src/test/java/org/pentaho/di/trans/steps/rest/RestTest.java [hunk 1]
```diff
@@ -77,6 +75,7 @@ public void testCreateMultivalueMap() {
     assertTrue( val1.contains( "%7D" ) );
   }
 
+  @Ignore
```
Comment (peterrinehart) [inline engine/src/test/java/org/pentaho/di/trans/steps/rest/RestTest.java:78]: Why are we ignoring this test now?

**Review by peterrinehart — CHANGES_REQUESTED** [review]

**Comment by buildguy** [comment]
### :warning: Build finished in 26m 30s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.metainject.MetaInjectMigrationTest.testMigration</b> (click to expand)</summary><p>

```
after the migration names that need migration should change to the new ones and all other should be kept the same expected:<DAT[ABASE_FIELD_NAME]> but was:<DAT[E_RANGE_START_FIELD]>
```

</p></details>

  
Tests run: **4082**, Failures: **1**, Skipped: **14** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-8924/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: engine/src/test/java/org/pentaho/di/trans/steps/rest/RestTest.java (hunk 1 above)
Comment (niehusa) [reply engine/src/test/java/org/pentaho/di/trans/steps/rest/RestTest.java:78]: Based on the comments left in the ticket, Apache Hop project was a fork from Pentaho.
They solved the same bug, but they move further to upgrade underlying libraries from 1.x to 2.x (jersey-client, etc), which we don't plan to do at this moment.

This PR **just** bring in changes in Rest.java. no library changes at all.

I got the idea to reference test code from Apache Hop, this test was marked `@Ignore`, so I assumed we can ignore for now, maybe wait till we do library upgrade and revisit it.

**Comment by buildguy** [comment]
### :warning: Build finished in 25m 7s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.metainject.MetaInjectMigrationTest.testMigration</b> (click to expand)</summary><p>

```
after the migration names that need migration should change to the new ones and all other should be kept the same expected:<DAT[ABASE_FIELD_NAME]> but was:<DAT[E_RANGE_START_FIELD]>
```

</p></details>

  
Tests run: **4082**, Failures: **1**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-8924/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by pdesai16 — APPROVED** [review]

**Comment by buildguy** [comment]
### :warning: Build finished in 22m 57s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.metainject.MetaInjectMigrationTest.testMigration</b> (click to expand)</summary><p>

```
after the migration names that need migration should change to the new ones and all other should be kept the same expected:<DAT[ABASE_FIELD_NAME]> but was:<DAT[E_RANGE_START_FIELD]>
```

</p></details>

  
Tests run: **4082**, Failures: **1**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-8924/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :warning: Build finished in 24m 35s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.metainject.MetaInjectMigrationTest.testMigration</b> (click to expand)</summary><p>

```
after the migration names that need migration should change to the new ones and all other should be kept the same expected:<DAT[ABASE_FIELD_NAME]> but was:<DAT[E_RANGE_START_FIELD]>
```

</p></details>

  
Tests run: **4082**, Failures: **1**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-8924/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :warning: Build finished in 23m 55s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.metainject.MetaInjectMigrationTest.testMigration</b> (click to expand)</summary><p>

```
after the migration names that need migration should change to the new ones and all other should be kept the same expected:<DAT[ABASE_FIELD_NAME]> but was:<DAT[E_RANGE_START_FIELD]>
```

</p></details>

  
Tests run: **4082**, Failures: **1**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-8924/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :warning: Build finished in 23m 11s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.metainject.MetaInjectMigrationTest.testMigration</b> (click to expand)</summary><p>

```
after the migration names that need migration should change to the new ones and all other should be kept the same expected:<DAT[ABASE_FIELD_NAME]> but was:<DAT[E_RANGE_START_FIELD]>
```

</p></details>

  
Tests run: **3971**, Failures: **1**, Skipped: **12** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-8924/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**